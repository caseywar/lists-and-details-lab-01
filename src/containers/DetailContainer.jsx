import React, { Component } from 'react';
import { findVillagerById } from '../services/AnimalCrossingApi';
import VillagerDetail from '../components/details/VillagerDetails';
import style from '../components/app/App.css'


export default class DetailContainer extends Component {
    state = {
        villagerObject: {},
        loading: true,
    };

    async componentDidMount() {
        this.setState({ loading: true });
        const villager = await findVillagerById(this.props.match.params.id);
        this.setState({
            villagerObject: villager,
            loading: false
        });
    }

    render() {

        const { villagerObject, loading } = this.state;
        if (loading) return <h1>Loading...</h1>;

        return(
            <div className={style.detailPage}>
                <h1>Villager Details</h1>
                <VillagerDetail villager={villagerObject} />
            </div>
        );
    }
};
