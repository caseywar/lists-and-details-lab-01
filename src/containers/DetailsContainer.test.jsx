import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import DetailContainer from './DetailContainer';

describe('Details Container test', () => {
    it('render villager details by id', async () => {
        render(
            <DetailContainer 
            match={{
                params: {
                    _id: '5f5fb4bbbfd05c2aed82e460'
                },
            }}
            />
        );

        const details = await screen.findByRole('villagerDetails');
        expect(details).toMatchSnapshot();
    })
})