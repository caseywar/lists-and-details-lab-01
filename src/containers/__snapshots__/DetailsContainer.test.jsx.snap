// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Container test render villager details by id 1`] = `
<main
  role="villagerDetails"
>
  <label />
  <img />
  <ul>
    <li />
    <li />
  </ul>
  <a
    href="/"
  >
    Back to List
  </a>
</main>
`;
